matrix_sdk/src/client.rs

```diff
@@ -828,6 +828,16 @@ impl Client {
         self.base_client.store()
     }
 
+    /// Store a value in the custom tree on the store
+    pub async fn store_save_value(&self, key: &[u8], value: Vec<u8>) -> Result<Option<Vec<u8>>> {
```

Comment:
As you noticed, you can get the `Store` from the client, so these methods are a bit redundant. I would say let's try to keep the `Client` slim, it is a bit an impossible task to do so considering the API surface of Matrix but removing this certainly would help with that.